app/controllers/declaration/DispatchLocationPageController.scala

```diff
@@ -45,12 +45,10 @@ class DispatchLocationPageController @Inject()(
 } with FrontendController(mcc) with I18nSupport with ModelCacheable with SessionIdAware {
 
   def displayPage(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    customsCacheService
-      .fetchAndGetEntry[DispatchLocation](cacheId, DispatchLocation.formId)
-      .map {
-        case Some(data) => Ok(dispatchLocationPage(DispatchLocation.form().fill(data)))
-        case _          => Ok(dispatchLocationPage(DispatchLocation.form()))
-      }
+    exportsCacheService.get(journeySessionId).map {
```

Comment:
Maybe  `.map(_.dispatchLocation)` before this map then `case  Right(Some(dispatchLocation))` below so we  don't need the `get`

Reply:
But it would  mean  you would end up on the empty form if the retrieve failed **or** returned nothing..... Might as well just use Option ;)